octodns/record/base.py

```diff
@@ -46,24 +46,33 @@ def new(cls, zone, name, data, source=None, lenient=False):
             reasons.append('invalid record, whitespace is not allowed')
 
         fqdn = f'{name}.{zone.name}' if name else zone.name
+        context = getattr(data, 'context', None)
```

Comment:
In contrast `Record` stuff may get data from our yaml infra or it may be coming from other providers that do not use `ContextDict` so we always need check to see if the attr exists gracefully, and then only use the context if there was a value (below.)